SETI/rms-opus
# Rename volumeid to bundleid

- Fixes #N/A
- Were any Django, import pipeline, or table_schema files modified? Y
  - If YES:
    - Database used (or new import): New import
    - FLAKE8 run on modified code: Y
    - All Django tests pass: Y
    - Code coverage run and still at 100%: Y
- Were any JavaScript or CSS files modified? Y
  - If YES:
    - JSHINT run on all affected files: Y
    - Tested on Chrome, Firefox, Safari, Edge, and iOS: NA
      (Remember to test all browser sizes)
    - Tested for race conditions (with delayed API calls if applicable): NA
- Was the documentation reviewed for necessary changes or additions? Y
  - About
  - Getting Started
  - FAQ
  - API Guide
  - Tooltips
  - Import/database schema

Description of changes:

This is the first set of changes required to move OPUS to supporting PDS4. It primarily renames `volume` to `bundle` everywhere to agree with the convention that everything should use PDS4 terminology going forward. The actual data imported has not changed. The UI adds new fields for PDS4, which are currently unpopulated, and renames some existing fields to be more general.

- Renamed volumeid to bundleid, volume to bundle everywhere. Note this means the API return values have changed! The previous `volumeid` return slug is now called `bundleid`.
- Kept `volumeid` as a backup slug name for `bundleid` to allow backwards compatibility
- Renamed UI "Volume ID" field to "Bundle/Volume ID"
- Renamed UI "Data Set ID" to "PDS3 Data Set ID"
- Renamed UI "Product ID" to "PDS3 Product ID"
- Added new field "Primary Product LID", which is currently None for all observations
- Update tooltips, help files, and help menu as necessary

Known problems:

- The descriptions of OPUS product types have not changed. Some of these, such as `rms_index` may need to be changed in the future to cover whatever we provide for PDS4 bundles.
